### ISSUE TITLE
group: fastest width in the west

### ISSUE THREAD

**Issue by lasers** [ultrabug/py3status#1062]
A wordplay on `Fastest Gun in the West`. :gun:
```
diff --git a/py3status/modules/bluetooth.py b/py3status/modules/bluetooth.py
index e4c22884..1402e3c0 100644
--- a/py3status/modules/bluetooth.py
+++ b/py3status/modules/bluetooth.py
@@ -124,10 +124,7 @@ class Py3status:
             self.py3.log(msg)

     def bluetooth(self):
-        try:
-            devices = get_connected_devices()
-        except dbus.DBusException:
-            self.py3.error(STRING_NOT_STARTED)
+        devices = get_connected_devices()

         if devices:
             data = []
diff --git a/py3status/modules/group.py b/py3status/modules/group.py
index 042b58a0..fcab964b 100644
--- a/py3status/modules/group.py
+++ b/py3status/modules/group.py
@@ -165,7 +165,9 @@ class Py3status:
             if not output:
                 widths.append(0)
             else:
-                widths.append(sum([len(x['full_text']) for x in output]))
+                now = sum([len(x['full_text']) for x in output])
+                widths.append(now)
+                self.py3.log('Now... %s' % now)
             if i == self.active:
                 current = output
                 current_width = widths[-1]

```
```
group uptime {
    fixed_width = True
    format = "{output}"

    bluetooth {}
    static_string a {format = 'ss_a'}
    static_string b {format = 'ss_b'}
    static_string c {format = 'ss_c'}
}
order += "group uptime"
```

Keep repeating this... Left click, scroll up... until you see it.

You can keep doing this and the width will continue to grow... It can go over 1000.

This could be @ultrabug's bug if he held down the middle button and scroll. However, this requires `fixed_width` to be enabled and he say it is not. I dunno. It could be the other things at the play here.

I'm still awaiting some information from him on `group` setting and other things. I'm pondering if this occurs on the laptop... Using touchpads... and/or its vertical/horizontal finger slides.... maybe if he's sporting some cool mouse toys... to scroll suddenly and/or forcefully... I don't know. I hope it's not too far-fetched.

Anyhow, we should find out if this is somehow related and/or is the underlying issue in the other bug... as I am trying to reproduce / hunt down @ultrabug's `group` bug. Many thanks. 🐞 🔫

**Comment by tobes** [comment]
@lasers thanks for looking at this issue.  This is a much better issue report 👍 although the `bluetooth` part of the patch I think may be un-needed.

From what I can see this needs bluetooth running which I don't have and may actually be about the `bluetooth` module.  If `bluetooth` output keeps increasing then `group` module is doing what it should, even if possibly not what is expected.

**Comment by lasers** [comment]
Nothing to do with `bluetooth`. I originally picked `bluetooth` long time ago because that one made a huge width exception... which may help with investigating the `group` width bug.

You should be able to add any other module and use `raise Exception('d')` in the `post_config_hook` to play with this particular bug.

**Comment by tobes** [comment]
Do we need an exception for the bug to happen?

**Comment by lasers** [comment]
I think so, yes. I can't reproduce it without an exception.

**Comment by tobes** [comment]
does this fix it?

```diff
diff --git a/py3status/py3.py b/py3status/py3.py
index 4b7af0c2..cb224f6e 100644
--- a/py3status/py3.py
+++ b/py3status/py3.py
@@ -5,6 +5,7 @@ import os
 import sys
 import shlex
 
+from copy import deepcopy
 from fnmatch import fnmatch
 from math import log10
 from pprint import pformat
@@ -471,7 +472,7 @@ class Py3:
         module_info = self._get_module_info(module_name)
         if module_info:
             output = module_info['module'].get_latest()
-        return output
+        return deepcopy(output)
 
     def trigger_event(self, module_name, event):
         """
```

it is because we end up changing the value in the module rather than our own private copy

**Comment by lasers** [comment]
I'll test it in an hour or so. Gotta go now.
![bug](https://user-images.githubusercontent.com/852504/30163438-9cbd4f7c-939e-11e7-947c-72a90aefdb7f.gif)

**Comment by lasers** [comment]
Good bugfix. 🏅

**Comment by lasers** [comment]
@tobes Make a PR for this one?

**Comment by lasers** [comment]
Closed via #1073.
